lib/aws_command.rb

```diff
@@ -23,12 +23,21 @@ def launch()
 
         # Expand all of the instance names so that we have a complete array
         names = []
-        options[:count].times { names << "#{options[:env]}-#{options[:type]}-#{SecureRandom.hex(5)}" }
+        minion_indexes = []
+        indexes_by_name = Hash.new
+        options[:count].times do |index|
+          name = "#{options[:env]}-#{options[:type]}-#{SecureRandom.hex(5)}"
+          names << name
+          minion_indexes << index
+          indexes_by_name[name] = "#{index}"
+        end
 
         ah = AnsibleHelper.for_aws()
 
         # AWS specific configs
         ah.extra_vars['oo_new_inst_names'] = names
+        ah.extra_vars['oo_indexes'] = minion_indexes
+	ah.extra_vars['oo_new_inst_indexes_by_name'] = indexes_by_name
```

Comment:
oo_new_inst_indexes_by_name doesn't appear to be used anywhere, looks like you can get rid of this, as well as the other references to indexes_by_name.